src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -87,12 +87,38 @@ public class GetAnalyzerSettings : Task
         public override bool Execute()
         {
             var config = TaskUtilities.TryGetConfig(AnalysisConfigDir, new MSBuildLoggerAdapter(Log));
-            ExecuteAnalysis(config);
+
+            if (ShouldMergeAnalysisSettings(config))
+            {
+                MergeAnalysisSettings(config);
+            }
+            else
+            {
+                OverrideAnalysisSettings(config);
+            }
 
             return !Log.HasLoggedErrors;
         }
 
-        private void ExecuteAnalysis(AnalysisConfig config)
+        internal /* for testing */ static bool ShouldMergeAnalysisSettings(AnalysisConfig config)
+        {
```

Comment:
Fortunately, the _SonarQubeAnalysisConfig.xml_ file that is produced by the _begin_ step already has the SonarQube version in it.